### PR TITLE
i18n(zh-cn): translate guides/actions.mdx into Chinese(Simplified)

### PR DESCRIPTION
**Pull request by CaptainOfPhB**
#### Description (required)

Translate the `/en/guides/actions` into Chinese(Simplified).

### CODE REVIEW

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+}
+```
+
+### 校验表单数据
+
+Actions 将解析提交的表单数据为一个对象，使用每个输入的 `name` 属性的值作为对象键。例如，包含 `<input name="search">` 的表单将被解析为一个对象，如 `{ search: 'user input' }`。你的 action 的 `input` 模式将用于验证此对象。
+
+为了接收原始的 `FormData` 对象，而不是解析后的对象，可以在 action 定义中省略 `input` 属性。
+
+下面的示例显示了一个验证过的 newsletter 注册表单，接受用户的电子邮件并要求“服务条款”复选框的同意。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:306]:
```suggestion
下面的示例显示了一个验证过的 newsletter 注册表单，它接受用户的电子邮件并要求勾选“服务条款”复选框以同意。
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+      form?.addEventListener('submit', async (event) => {
+        event.preventDefault();
+        const formData = new FormData(form);
+        const { error } = await actions.newsletter(formData);
+        if (!error) navigate('/confirmation');
+      })
+    </script>
+    ```
+
+    <ReadMore>请参阅 [“从 HTML 表单操作调用 action”](#从-html-表单操作调用-action) 了解提交表单数据的另一种方法</ReadMore>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:371]:
```suggestion
    <ReadMore>请参阅 [“从 HTML 表单操作调用 action”](#从-html-表单操作调用-action) 以了解提交表单数据的另一种方法。</ReadMore>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+
+</Steps>
+
+### 展示表单输入错误
+
+你可以在提交前校验表单输入，使用原生 HTML 表单校验属性，例如 `required`、`type="email"` 和 `pattern`。对于后端的更复杂的 `input` 校验，你可以使用 [`isInputError()`](/zh-cn/reference/api-reference/#isinputerror) 工具函数。
+
+为了检索输入错误，使用 `isInputError()` 工具函数检查错误是否由无效输入引起。输入错误包含一个 `fields` 对象，其中包含每个未能验证的输入名称的消息。你可以使用这些消息提示用户纠正他们的提交。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:379]:
```suggestion
要检索输入错误，可以使用 `isInputError()` 工具函数来检查错误是否是由无效输入而引起的。输入错误包含一个 `fields` 对象，其中包含每个验证失败的输入名称的消息。你可以使用这些消息提示用户以纠正他们的提交。
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+下面的示例使用 `isInputError()` 检查错误，然后检查错误是否在电子邮件字段中，最后从错误中创建消息。你可以使用 JavaScript DOM 操作或你喜欢的 UI 框架将此消息显示给用户。
+
+```js /isInputError(?= )/ {5-12}
+import { actions, isInputError } from 'astro:actions';
+
+const form = document.querySelector('form');
+const formData = new FormData(form);
+const { error } = await actions.newsletter(formData);
+if (isInputError(error)) {
+  // Handle input errors.
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:390]:
```suggestion
  // 处理输入错误。
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+
+:::note
+页面必须在使用表单操作调用 actions 时进行按需渲染。在使用此 API 之前，请[确保页面上已禁用预渲染](/zh-cn/guides/server-side-rendering/#在-hybrid-模式中选择退出预渲染)。
+:::
+
+你可以在任何 `<form>` 元素上使用标准属性启用零 JS 表单提交。无需客户端 JavaScript 的表单提交可能对于 JavaScript 加载失败时的回退，或者如果你更喜欢完全从服务器处理表单时很有用。
+
+调用 [Astro.getActionResult()](/zh-cn/reference/api-reference/#astrogetactionresult) 在服务器上返回表单提交的结果（`data` 或 `error`），可以用于动态重定向、处理表单错误、更新 UI 等。
+
+调用一个 action 时，Astro 会自动处理表单提交，并将结果返回给你的页面。这使得你可以在表单提交后更新 UI，重定向到一个新页面，或者处理任何错误。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:407]:
```suggestion
你可以在任何 `<form>` 元素上使用标准属性启用零 JS 表单提交。无论是作为 JavaScript 加载失败时的回退，又或者你更喜欢仅从服务器来处理表单时，无需客户端 JavaScript 的表单提交都非常有用。

在服务器上调用 [Astro.getActionResult()](/zh-cn/reference/api-reference/#astrogetactionresult) 会返回表单提交的结果（`data` 或 `error`），并且可以用于动态重定向、处理表单错误、更新 UI 等。

要从 HTML 表单调用 action，可以为 `<form>` 添加 `method="POST"`，并设置表单的 `action` 属性使用你的 action，例如 `action={actions.logout}`。这会将 `action` 属性设置为使用服务器自动处理的查询字符串。
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 6]
```diff
@@ -0,0 +1,580 @@
...
+---
+
+<form method="POST" action={actions.logout}>
+  <button>Log out</button>
+</form>
+```
+
+### 在 action 成功时重定向
+
+为了在一个 action 成功时导航到一个不同的页面，你可以在 `action` 属性中添加一个路径。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:423]:
```suggestion
要在没有客户端 JavaScript 且 action 成功时导航到一个不同的页面，你可以在 `action` 属性中添加一个路径。
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 7]
```diff
@@ -0,0 +1,580 @@
...
+  return Astro.redirect(`/products/${result.data.id}`);
+}
+---
+
+<form method="POST" action={actions.createProduct}>
+  <!--...-->
+</form>
+```
+
+### 处理表单操作错误
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:479]:
```suggestion
### 处理表单 action 错误
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 8]
```diff
@@ -0,0 +1,580 @@
...
+
+输入框在提交表单时会被清空。为了保留输入值，你可以[在页面上启用视图过渡](/zh-cn/guides/view-transitions/#向页面添加视图过渡动画)，并对每个输入应用 `transition:persist` 指令：
+
+```astro ins="transition:persist"
+<input transition:persist required type="email" name="email" />
+```
+
+### 使用表单操作结果更新 UI
+
+由 `Astro.getActionResult()` 返回的结果是一次性的，每当页面刷新时都会重置为 `undefined`。这对于[处理表单操作错误](#处理表单操作错误)和在成功时向用户显示临时通知非常理想。
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:540]:
```suggestion
### 使用表单的 action 结果以更新 UI

由 `Astro.getActionResult()` 返回的结果是一次性的，每当页面刷新时都会重置为 `undefined`。这对于[处理表单操作错误](#处理表单-action-错误)和在成功时向用户显示临时通知非常理想。
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 9]
```diff
@@ -0,0 +1,580 @@
...
+    import { defineAction } from 'astro:actions';
+    import { z } from 'astro:schema';
+
+    export const server = {
+      getGreeting: defineAction({
+        input: z.object({
+          name: z.string(),
+        }),
+        handler: async (input) => {
+          return `Hello, ${input.name}!`
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:88]:
```suggestion
          return `你好，${input.name}!`
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 10]
```diff
@@ -0,0 +1,580 @@
...
+    ```astro title="src/pages/index.astro"
+    ---
+    ---
+
+    <button>Get greeting</button>
+
+    <script>
+    const button = document.querySelector('button');
+    button?.addEventListener('click', async () => {
+      // Show alert pop-up with greeting from action
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:105]:
```suggestion
    <button>获取问候语</button>

    <script>
    const button = document.querySelector('button');
    button?.addEventListener('click', async () => {
      // 通过 action 弹出带有问候语的弹窗
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 11]
```diff
@@ -0,0 +1,580 @@
...
+    ---
+
+    <button>Get greeting</button>
+
+    <script>
+    import { actions } from 'astro:actions';
+
+    const button = document.querySelector('button');
+    button?.addEventListener('click', async () => {
+      // Show alert pop-up with greeting from action
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:123]:
```suggestion
    <button>获取问候语</button>

    <script>
    import { actions } from 'astro:actions';

    const button = document.querySelector('button');
    button?.addEventListener('click', async () => {
      // 通过 action 弹出带有问候语的弹窗
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 12]
```diff
@@ -0,0 +1,580 @@
...
+```tsx title=src/components/LikeButton.tsx ins="if (error.code === 'UNAUTHORIZED') setShowLogin(true);"
+import { actions } from 'astro:actions';
+import { useState } from 'preact/hooks';
+
+export function LikeButton({ postId }: { postId: string }) {
+  const [showLogin, setShowLogin] = useState(false);
+  return (
+    <>
+      {
+        showLogin && <a href="/signin">Log in to like a post.</a>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:245]:
```suggestion
        showLogin && <a href="/signin">登录后点赞文章。</a>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 13]
```diff
@@ -0,0 +1,580 @@
...
+      }
+      <button onClick={async () => {
+        const { data, error } = await actions.likePost({ postId });
+        if (error?.code === 'UNAUTHORIZED') setShowLogin(true);
+        // Early return for unexpected errors
+        else if (error) return;
+        // update likes
+      }}>
+        Like
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:254]:
```suggestion
        // 因意外错误而提前终止
        else if (error) return;
        // 更新点赞数
      }}>
        点赞
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 14]
```diff
@@ -0,0 +1,580 @@
...
+import { actions } from 'astro:actions';
+import { navigate } from 'astro:transitions/client';
+
+export function LogoutButton() {
+  return (
+    <button onClick={async () => {
+      const { error } = await actions.logout();
+      if (!error) navigate('/');
+    }}>
+      Logout
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:277]:
```suggestion
      退出登录
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 15]
```diff
@@ -0,0 +1,580 @@
...
+
+    ```astro title="src/components/Newsletter.astro" /name="\w+"/
+    <form>
+      <label for="email">E-mail</label>
+      <input id="email" required type="email" name="email" />
+      <label>
+        <input required type="checkbox" name="terms">
+        I agree to the terms of service
+      </label>
+      <button>Sign up</button>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:320]:
```suggestion
        我已同意服务条款
      </label>
      <button>注册</button>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 16]
```diff
@@ -0,0 +1,580 @@
...
+
+    ```astro title=src/components/Newsletter.astro ins={12-23} collapse={2-8}
+    <form>
+      <label for="email">E-mail</label>
+      <input id="email" required type="email" name="email" />
+      <label>
+        <input required type="checkbox" name="terms">
+        I agree to the terms of service
+      </label>
+      <button>Sign up</button>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:354]:
```suggestion
        我已同意服务条款
      </label>
      <button>注册</button>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 17]
```diff
@@ -0,0 +1,580 @@
...
+
+例如，这个 Astro 组件在点击按钮时调用 `logout` action 并重新加载当前页面：
+
+```astro title="src/components/LogoutButton.astro"
+---
+import { actions } from 'astro:actions';
+---
+
+<form method="POST" action={actions.logout}>
+  <button>Log out</button>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:417]:
```suggestion
  <button>退出登录</button>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 18]
```diff
@@ -0,0 +1,580 @@
...
+例如，当 `newsletter` action 成功时，`action={'/confirmation' + actions.newsletter}` 将导航到 `/confirmation`：
+
+```astro title="src/components/NewsletterSignup.astro" /action=\{[^\{\}]+\}/
+---
+import { actions } from 'astro:actions';
+---
+
+<form method="POST" action={'/confirmation' + actions.newsletter}>
+  <label>E-mail <input required type="email" name="email" /></label>
+  <button>Sign up</button>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:434]:
```suggestion
  <button>注册</button>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 19]
```diff
@@ -0,0 +1,580 @@
...
+
+{result?.error && (
+  <p class="error">Unable to sign up. Please try again later.</p>
+)}
+<form method="POST" action={'/confirmation' + actions.newsletter}>
+  <label>
+    E-mail
+    <input required type="email" name="email" />
+  </label>
+  <button>Sign up</button>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:500]:
```suggestion
  <p class="error">无法注册。请稍后再试。</p>
)}
<form method="POST" action={'/confirmation' + actions.newsletter}>
  <label>
    E-mail
    <input required type="email" name="email" />
  </label>
  <button>注册</button>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 20]
```diff
@@ -0,0 +1,580 @@
...
+const inputErrors = isInputError(result?.error) ? result.error.fields : {};
+---
+
+<form method="POST" action={'/confirmation' + actions.newsletter}>
+  <label>
+    E-mail
+    <input required type="email" name="email" aria-describedby="error" />
+  </label>
+  {inputErrors.email && <p id="error">{inputErrors.email.join(',')}</p>}
+  <button>Sign up</button>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:522]:
```suggestion
  <button>注册</button>
```

File: src/content/docs/zh-cn/guides/actions.mdx [hunk 21]
```diff
@@ -0,0 +1,580 @@
...
+
+```astro title="src/pages/products/[slug].astro"
+---
+import { actions } from 'astro:actions';
+
+const result = Astro.getActionResult(actions.addToCart);
+---
+
+{result && !result.error && (
+  <p class="success">Added {result.data.productName} to cart</p>
```
Comment (Nin3lee) [inline src/content/docs/zh-cn/guides/actions.mdx:556]:
```suggestion
  <p class="success">添加 {result.data.productName} 到购物车</p>
```

**Review by Nin3lee — CHANGES_REQUESTED** [review]

**Review by liruifengv — APPROVED** [review]
LGTM! Thank you all! @CaptainOfPhB @Nin3lee

**Pull request merged**